### PR TITLE
Websocket Streaming Interface

### LINKED ISSUE #2135: Feature: swap out octant's streaming (websockets) with another implementation
**Describe the problem/challenge you have**
I would like the ability to inject my own streaming solution into octant instead of being tied to the websocket implementation. For instance if I wanted to experiment with grpc-web, or have a service that translates grpc to websockets

**Describe the solution you'd like**
The ability to swap out the implementation of octant's streaming implementation

ref: tdc

### PR DESCRIPTION
**Pull request by wwitzel3**
**What this PR does / why we need it**:
Replaces #2504 and #1905

**Which issue(s) this PR fixes**
- Fixes #2135

### CODE REVIEW

**Comment by wwitzel3** [comment]
I think this is now in it's final form @mstergianis @jamieklassen 

I've rebased, made some minor refactors, and also ensured all of the Michael's recent changes in his branch have been added in as well.

File: internal/api/helper_manager.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type HelperStateManager struct {
 	poller             Poller
 }
 
-var _StateManager = (*HelperStateManager)(nil)
+var _ = (*HelperStateManager)(nil)
```
Comment (mstergianis) [inline internal/api/helper_manager.go:47]:
```suggestion
var _ StateManager = (*HelperStateManager)(nil)
```

**Comment by wwitzel3** [comment]
@GuessWhoSamFoo could use a review on this.

**Review by GuessWhoSamFoo — APPROVED** [review]
Depending on merge order,  this one or #2566 will need a rebase

**Comment by zoetian** [comment]
Hey @GuessWhoSamFoo , is it possible to get this one in first please? =) 
Some of our work is currently blocked by this MR and it would be really helpful if it could be merged. 
Many many thanks!

**Comment by GuessWhoSamFoo** [comment]
The other PR is a dependency update with regenerated mock files so it should not block (and trivial to rebase if we want both)

**Pull request merged**